src/gtktest.rs

```diff
@@ -3,295 +3,213 @@
-            Err(_) => Some(Err(())),
-        }
-    });
-
-    let entry_clone = entry.clone();
-    button.connect_clicked(clone!(window => move |_| {
-        let dialog = gtk::Dialog::new_with_buttons(Some("Hello!"), Some(&window), gtk::DIALOG_MODAL,
-            &[("No", 0), ("Yes", 1), ("Yes!", 2)]);
-
-        let ret = dialog.run();
-
-        dialog.destroy();
-
-        entry_clone.set_text(&format!("Clicked {}", ret));
-    }));
-
-    // use a plain function instead of a closure
-    button_about.connect_clicked(about_clicked);
-
-    button_font.connect_clicked(clone!(window => move |_| {
-        let dialog = gtk::FontChooserDialog::new(Some("Font chooser test"), Some(&window));
+        );
+    }
 
-        dialog.run();
-        dialog.destroy();
-    }));
+    fn about_clicked(button: &gtk::Button, builder: &Builder) {
+        let dialog: gtk::AboutDialog = builder.get_object("dialog").unwrap();
+        if let Some(window) = button.get_toplevel().and_then(|w| w.downcast::<gtk::Window>().ok()) {
+            dialog.set_transient_for(Some(&window));
+        }
 
-    button_recent.connect_clicked(clone!(window => move |_| {
-        let dialog = gtk::RecentChooserDialog::new(Some("Recent chooser test"), Some(&window));
-        dialog.add_buttons(&[
-            ("Ok", gtk::ResponseType::Ok.into()),
-            ("Cancel", gtk::ResponseType::Cancel.into())
-        ]);
+        println!("Authors: {:?}", dialog.get_authors());
+        println!("Artists: {:?}", dialog.get_artists());
+        println!("Documenters: {:?}", dialog.get_documenters());
 
         dialog.run();
         dialog.destroy();
-    }));
-
-    file_button.connect_clicked(clone!(window => move |_| {
-        //entry.set_text("Clicked!");
-        let dialog = gtk::FileChooserDialog::new(Some("Choose a file"), Some(&window),
-            gtk::FileChooserAction::Open);
-        dialog.add_buttons(&[
-            ("Open", gtk::ResponseType::Ok.into()),
-            ("Cancel", gtk::ResponseType::Cancel.into())
-        ]);
-
-        dialog.set_select_multiple(true);
-        dialog.run();
-        let files = dialog.get_filenames();
-        dialog.destroy();
-
-        println!("Files: {:?}", files);
-    }));
+    }
 
-    app_button.connect_clicked(clone!(window => move |_| {
-        //entry.set_text("Clicked!");
-        let dialog = gtk::AppChooserDialog::new_for_content_type(Some(&window), gtk::DIALOG_MODAL,
-            "sh");
+    pub fn sub_main() {
+        if gtk::init().is_err() {
+            println!("Failed to initialize GTK.");
+            return;
+        }
+        println!("Major: {}, Minor: {}", gtk::get_major_version(), gtk::get_minor_version());
+        let glade_src = include_str!("gtktest.glade");
+        let builder = Builder::new_from_string(glade_src);
+
+        let window: gtk::Window = builder.get_object("window").unwrap();
+        let button: gtk::Button = builder.get_object("button").unwrap();
```

Comment:
Better import `Window`, `Button` and so on. Also the code would probably be easier to understand if calls to `connect` were closer to `get_object`. This `button` is used like 30 lines from where it's created.